### PR TITLE
Adds support for removing selectors

### PR DESCRIPTION
**Pull request by schuylermartin45**
- `remove_selectors()` allows for the removal of selectors on a parse tree path
- The function preserves non-selector comments
- Adds unit tests
- Adds some unrelated unit testing around `__eq__()` and `__str__()` functions that got missed.

### CODE REVIEW

File: percy/tests/test_aux_files/simple-recipe.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ requirements:
   empty_field1:
   host:
     - setuptools  # [unix]
-    - fakereq  # [unix]
+    - fakereq  # [unix] selector with comment
```
Comment (skupr-anaconda) [inline percy/tests/test_aux_files/simple-recipe.yaml:18]: How about this usage:
`- posix    # [win] # a real comment`

Will this edge case pass or fail? It's not so often but it can happen in the meta.yaml files

File: percy/render/recipe_parser.py [hunk 2]
```diff
@@ -1433,6 +1449,40 @@ def _extract_selector(s: str) -> str:
...
+
+        node.comment = comment
+        # Some lines of YAML correspond to multiple nodes. For consistency,
+        # we need to ensure that comments are duplicate across all nodes on a line.
+        if node.is_single_key():
+            node.children[0].comment = comment
+
+        self._rebuild_selectors()
+        self._is_modified = True
+
```
Comment (cbouss) [inline percy/render/recipe_parser.py:1485]: You could make that function return the selector string that was removed.

File: percy/tests/test_aux_files/simple-recipe.yaml (hunk 1 above)
Comment (schuylermartin45) [reply percy/tests/test_aux_files/simple-recipe.yaml:18]: A quick and dirty test shows:
```
    - fakereq  # [unix] # selector with comment
```
becomes
```
    - fakereq  # selector with comment
```
So yes, it works. I can add this variant as a test case.

Comment (schuylermartin45) [reply percy/tests/test_aux_files/simple-recipe.yaml:18]: Correction: this worked for the wrong reasons. There was a previously unknown issue parsing comments with `#` symbols in the comments that made this "work". 

A forthcoming commit will resolve the bug and this issue.

Comment (skupr-anaconda) [reply percy/tests/test_aux_files/simple-recipe.yaml:18]: Looks good

**Review by Arishamays1 — APPROVED** [review]
LGTM
